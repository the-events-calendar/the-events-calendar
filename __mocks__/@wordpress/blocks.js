import noop from 'lodash/noop';

export const createBlock = noop;
