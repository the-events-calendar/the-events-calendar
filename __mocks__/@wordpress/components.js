/**
 * External dependencies
 */
import { noop } from 'lodash';

export const withAPIData = () => noop;
export const Spinner = () => "🏃‍♂️";
export const Modal = ( { title, children } ) => (
	<div>
		<span>{ title }</span>
		<span>{ children }</span>
	</div>
);
export const Dashicon = ( { className, icon } ) => <span className={ className }>{ icon }</span>;
export const Dropdown = () => <span>Dropdown</span>;
export const Tooltip = () => <span>Tooltip</span>;
export const PanelBody = ({ children }) => <span className="PanelBody">{ children }</span>
