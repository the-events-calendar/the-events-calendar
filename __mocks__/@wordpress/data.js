/**
 * External dependencies
 */
import { noop } from 'lodash';
import { Component } from 'react';

export const select = noop;
export const withSelect = () => ( component ) => component;
export const withDispatch = () => ( component ) => component;
