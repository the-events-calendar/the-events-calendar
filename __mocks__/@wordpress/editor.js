/**
 * External dependencies
 */
import React from 'react';

export const RichText = () => <div>Rich Text</div>;
export const InspectorControls = ({ children }) => <div className="InspectorControls">{ children }</div>
