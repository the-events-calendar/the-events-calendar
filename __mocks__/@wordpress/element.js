/**
 * External dependencies
 */
import Element, { Component } from 'react';
import { compose } from 'redux';

export {
	Element,
	Component,
	compose,
};
