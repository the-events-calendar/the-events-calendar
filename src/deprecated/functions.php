<?php
/**
 * register a meta group
 *
 * @uses Tribe__Events__Meta_Factory::register()
 * @deprecated 4.3
 *
 * @param string $meta_group_id
 * @param array  $args
 *
 * @return bool $success
 */
function tribe_register_meta_group( $meta_group_id, $args = [] ) {
	_deprecated_function( __FUNCTION__, '4.3' );

	// setup default for registering a meta group
	$defaults = [ 'register_type' => 'meta_group', 'register_overwrite' => true ];

	// parse the $default and $args into the second param for registering a meta item
	return Tribe__Events__Meta_Factory::register( $meta_group_id, wp_parse_args( $args, $defaults ) );
}

/**
 * register a meta item
 *
 * @uses Tribe__Events__Meta_Factory::register()
 * @deprecated 4.3
 *
 * @param int   $meta_id
 * @param array $args
 *
 * @return bool $success
 */
function tribe_register_meta( $meta_id, $args = [] ) {
	_deprecated_function( __FUNCTION__, '4.3' );

	return Tribe__Events__Meta_Factory::register( $meta_id, $args );
}

/**
 * Get the meta group.
 *
 * @deprecated 4.3
 *
 * @param      $meta_group_id
 * @param bool $is_the_meta
 *
 * @return bool|mixed|void
 */
function tribe_get_meta_group( $meta_group_id, $is_the_meta = false ) {
	_deprecated_function( __FUNCTION__, '4.3' );

	do_action( 'tribe_get_meta_group', $meta_group_id, $is_the_meta );

	$type = 'meta_group';

	// die silently if the requested meta group is not registered
	if ( ! Tribe__Events__Meta_Factory::check_exists( $meta_group_id, $type ) ) {
		return false;
	}

	$meta_group = Tribe__Events__Meta_Factory::get_args( $meta_group_id, $type );
	$meta_ids   = Tribe__Events__Meta_Factory::get_order( $meta_group_id );
	$group_html = '';

	// internal check for hiding items in the meta
	if ( ! $meta_group['show_on_meta'] ) {
		return false;
	}
	$meta_pos_int     = 0;
	$total_meta_items = tribe_count_hierarchical( $meta_ids );
	foreach ( $meta_ids as $meta_id_group ) {
		foreach ( $meta_id_group as $meta_id ) {
			$meta_pos_int ++;

			$group_html = tribe_separated_field( $group_html, $meta_group['wrap']['meta_separator'], tribe_get_meta( $meta_id, $is_the_meta ) );
		}
	}

	$params = [ $meta_group_id ];

	if ( ! empty( $meta['filter_callback'] ) ) {
		return call_user_func_array( $meta['filter_callback'], $params );
	}

	if ( ! empty( $meta['callback'] ) ) {
		$value = call_user_func_array( $meta['callback'], $params );
	}

	$value = empty( $value ) ? $group_html : $value;

	$html = ! empty( $group_html ) ? Tribe__Events__Meta_Factory::template( $meta_group['label'], $value, $meta_group_id, 'meta_group' ) : '';

	return apply_filters( 'tribe_get_meta_group', $html, $meta_group_id );
}

/**
 * Get the meta.
 *
 * @deprecated 4.3
 *
 * @param      $meta_id
 * @param bool $is_the_meta
 *
 * @return bool|mixed|void
 */
function tribe_get_meta( $meta_id, $is_the_meta = false ) {
	_deprecated_function( __FUNCTION__, '4.3' );

	do_action( 'tribe_get_meta', $meta_id, $is_the_meta );

	// die silently if the requested meta item is not registered
	if ( ! Tribe__Events__Meta_Factory::check_exists( $meta_id ) ) {
		return false;
	}

	$meta = Tribe__Events__Meta_Factory::get_args( $meta_id );

	// internal check for hiding items in the meta
	if ( ! $meta['show_on_meta'] ) {
		return false;
	}

	$params = [ $meta_id ];

	if ( ! empty( $meta['filter_callback'] ) ) {
		return call_user_func_array( $meta['filter_callback'], $params );
	}

	if ( ! empty( $meta['callback'] ) ) {
		$value = call_user_func_array( $meta['callback'], $params );
	}

	$value = empty( $value ) ? $meta['meta_value'] : $value;

	// if we have a value let's build the html template
	$html = ! empty( $value ) ? Tribe__Events__Meta_Factory::template( $meta['label'], $value, $meta_id ) : '';

	return apply_filters( 'tribe_get_meta', $html, $meta_id );
}

/**
 * Get the args for a meta object.
 *
 * @deprecated 4.3
 *
 * @param        $meta_id
 * @param        $arg_key
 * @param string $type
 *
 * @return bool
 */
function tribe_get_meta_arg( $meta_id, $arg_key, $type = 'meta' ) {
	_deprecated_function( __FUNCTION__, '4.3' );

	// die silently if the requested meta group is not registered
	if ( ! Tribe__Events__Meta_Factory::check_exists( $meta_id, $type ) ) {
		return false;
	}

	$args = Tribe__Events__Meta_Factory::get_args( $meta_id, $type );

	// check if the arg exists
	if ( isset( $args[ $arg_key ] ) ) {
		return $args[ $arg_key ];
	} else {
		return false;
	}
}

/**
 * Get the template part for the meta object.
 *
 * @deprecated 4.3
 *
 * @param        $meta_id
 * @param        $template_key
 * @param string $type
 *
 * @return bool
 */
function tribe_get_meta_template_part( $meta_id, $template_key, $type = 'meta' ) {
	_deprecated_function( __FUNCTION__, '4.3' );

	// die silently if the requested meta group is not registered
	if ( ! Tribe__Events__Meta_Factory::check_exists( $meta_id, $type ) ) {
		return false;
	}

	$template = tribe_get_meta_arg( $meta_id, 'wrap', $type );

	if ( isset( $template[ $template_key ] ) ) {
		return $template[ $template_key ];
	} else {
		return false;
	}
}

/**
 * Set the visibility of the meta object.
 *
 * @deprecated 4.3
 *
 * @param        $meta_id
 * @param bool   $status
 * @param string $type
 */
function tribe_set_the_meta_visibility( $meta_id, $status = true, $type = 'meta' ) {
	_deprecated_function( __FUNCTION__, '4.3' );

	Tribe__Events__Meta_Factory::set_visibility( $meta_id, $type, $status );
}

/**
 * Set the template for the meta object.
 *
 * @deprecated 4.3
 *
 * @param        $meta_id
 * @param array  $template
 * @param string $type
 *
 * @return bool
 */
function tribe_set_the_meta_template( $meta_id, $template = [], $type = 'meta' ) {
	_deprecated_function( __FUNCTION__, '4.3' );

	if ( is_array( $meta_id ) ) {
		foreach ( $meta_id as $id ) {
			tribe_set_the_meta_template( $id, $template, $type );
		}
	} else {
		global $_tribe_meta_factory;

		// die silently if the requested meta group is not registered
		if ( ! Tribe__Events__Meta_Factory::check_exists( $meta_id, $type ) ) {
			return false;
		}

		if ( ! empty( $template ) ) {
			$_tribe_meta_factory->{$type}[ $meta_id ]['wrap'] = wp_parse_args( $template, $_tribe_meta_factory->{$type}[ $meta_id ]['wrap'] );
		}
	}

}

/**
 * Set the meta priority to manage positioning.
 *
 * @deprecated 4.3
 *
 * @param        $meta_id
 * @param int    $priority
 * @param string $type
 *
 * @return bool
 */
function tribe_set_meta_priority( $meta_id, $priority = 100, $type = 'meta' ) {
	_deprecated_function( __FUNCTION__, '4.3' );

	if ( is_array( $meta_id ) ) {
		foreach ( $meta_id as $id => $priority ) {
			tribe_set_meta_priority( $id, $priority, $type );
		}
	} else {
		global $_tribe_meta_factory;

		// die silently if the requested meta group is not registered
		if ( ! Tribe__Events__Meta_Factory::check_exists( $meta_id, $type ) ) {
			return false;
		}

		if ( ! empty( $priority ) ) {
			$_tribe_meta_factory->{$type}[ $meta_id ]['priority'] = $priority;
		}
	}
}

/**
 * Set meta value for meta object.
 *
 * @deprecated 4.3
 *
 * @param        $meta_id
 * @param        $value
 * @param string $value_type
 * @param string $type
 *
 * @return bool
 */
function tribe_set_meta_value( $meta_id, $value, $value_type = 'meta_value', $type = 'meta' ) {
	_deprecated_function( __FUNCTION__, '4.3' );

	if ( is_array( $meta_id ) ) {
		foreach ( $meta_id as $id ) {
			tribe_set_meta_value( $id, $value, $value_type, $type );
		}
	} else {
		global $_tribe_meta_factory;

		// die silently if the requested meta group is not registered
		if ( ! Tribe__Events__Meta_Factory::check_exists( $meta_id, $type ) ) {
			return false;
		}

		$_tribe_meta_factory->{$type}[ $meta_id ][ $value_type ] = $value;
	}
}

/**
 * Set the meta label for the meta object.
 *
 * @deprecated 4.3
 *
 * @param        $meta_id
 * @param string $label
 * @param string $type
 *
 * @return bool
 */
function tribe_set_meta_label( $meta_id, $label = '', $type = 'meta' ) {
	_deprecated_function( __FUNCTION__, '4.3' );

	if ( is_array( $meta_id ) ) {
		foreach ( $meta_id as $id => $label ) {
			tribe_set_meta_label( $id, $label, $type );
		}
	} else {
		global $_tribe_meta_factory;

		// die silently if the requested meta group is not registered
		if ( ! Tribe__Events__Meta_Factory::check_exists( $meta_id, $type ) ) {
			return false;
		}

		$_tribe_meta_factory->{$type}[ $meta_id ]['label'] = $label;
	}
}

/**
 * Get the event meta
 *
 * @deprecated 4.3
 *
 * @return mixed|void
 */
function tribe_get_the_event_meta() {
	_deprecated_function( __FUNCTION__, '4.3' );

	$html = '';
	foreach ( Tribe__Events__Meta_Factory::get_order() as $meta_groups ) {
		foreach ( $meta_groups as $meta_group_id ) {
			$html .= tribe_get_meta_group( $meta_group_id, true );
		}
	}

	return apply_filters( 'tribe_get_the_event_meta', $html );
}

/**
 * Simple display of meta group tag
 *
 * @deprecated 4.3
 *
 * @uses tribe_get_meta_group()
 * @return echo tribe_get_meta_group( $meta_group_id )
 */
function tribe_display_the_event_meta() {
	_deprecated_function( __FUNCTION__, '4.3' );

	echo apply_filters( 'tribe_display_the_event_meta', tribe_get_the_event_meta() );
}

/**
 *  Simple diplay of meta group tag
 *
 * @uses tribe_get_meta_group()
 * @deprecated 4.3
 *
 * @param string $meta_group_id
 *
 * @return echo tribe_get_meta_group( $meta_group_id )
 */
function tribe_display_meta_group( $meta_group_id ) {
	_deprecated_function( __FUNCTION__, '4.3' );

	echo apply_filters( 'tribe_display_meta_group', tribe_get_meta_group( $meta_group_id ) );
}

/**
 *  Simple diplay of meta tag
 *
 * @uses tribe_get_meta()
 * @deprecated 4.3
 *
 * @param string $meta_id
 *
 * @return echo tribe_get_meta( $meta_id )
 */
function tribe_display_meta( $meta_id ) {
	_deprecated_function( __FUNCTION__, '4.3' );

	echo apply_filters( 'tribe_display_meta', tribe_get_meta( $meta_id ) );
}

/**
 *  Load Legacy Imports
 *
 * @deprecated 4.6.18
 */
if ( ! function_exists( 'Tribe_Events_Importer_Load' ) ) {

	function Tribe_Events_Importer_Load() {
		_deprecated_function( __FUNCTION__, '4.5' );

		Tribe__Events__Importer__Plugin::set_plugin_basename( plugin_basename( __FILE__ ) );
		if ( is_admin() ) {
			add_action( 'init', [ 'Tribe__Events__Importer__Plugin', 'initialize_admin' ], 10, 0 );
			add_action( 'init', [ 'Tribe__Events__Importer__Options', 'process_general_form_submission' ] );
		}
	}

}
