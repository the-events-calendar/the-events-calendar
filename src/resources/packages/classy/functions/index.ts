export * as events from './events';
export * as date from './date';
