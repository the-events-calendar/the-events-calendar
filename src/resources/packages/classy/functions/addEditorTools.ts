import { __ } from '@wordpress/i18n';
import { registerPlugin } from '@wordpress/plugins';

/**
 * Adds the Editor tools inserting them among the Block Editor tools.
 *
 * This function is weird-looking to work around the lack of an official API to add/remove Editor tools.
 * The function registers a plugin to be called when the toolbar is rendered; then calls a function that looks like
 * a React Component (EditorTools) that will not actually render a component, but will use the call to insert a button
 * in the Editor tools.
 *
 * @since TBD
 *
 * @param {(this: GlobalEventHandlers, ev: MouseEvent) => void} onClick The click event to handle on press of the added
 *     button.
 * @param {Document|null} document The document to work on, or `null` to work on `window.document`.
 *
 * @return {void} The Editor tools are added to the tools bar.
 */
export function addEditorTools(
	onClick: ( this: GlobalEventHandlers, ev: MouseEvent ) => void,
	document: Document | null = null
): void {
	document = document || window.document;

	// 2. The function _should_ render a React component, but won't. It will insert a button.
	function EditorTools() {
		const editorDocumentTools = document.querySelector(
			'.editor-document-tools .editor-document-tools__left'
		);
		const previewButton = document.querySelector(
			'.tec-editor-tool--preview'
		);

		if ( editorDocumentTools && previewButton === null ) {
			const previewButton = document.createElement( 'button' );
			previewButton.classList.add(
				'tec-editor-tool',
				'tec-editor-tool--preview',
				'button'
			);
			previewButton.type = 'button';
			previewButton.dataset.toolbarItem = 'true';
			previewButton.innerHTML = `<span class="dashicons dashicons-visibility"></span>${ __(
				'Visual',
				'the-events-calendar'
			) }`;
			editorDocumentTools.append( previewButton );
			previewButton.onclick = onClick;
		}

		// 3. Do not actually render any component.
		return null;
	}

	// 1. When the time comes for plugin registration, it's the moment to insert the button.
	registerPlugin( 'tec-editor-tools', {
		render: EditorTools,
	} );
}
