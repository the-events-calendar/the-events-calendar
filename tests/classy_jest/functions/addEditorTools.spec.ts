import { addEditorTools } from '../../../src/resources/packages/classy/functions/addEditorTools';
import { afterEach, describe, it, expect, jest } from '@jest/globals';

jest.mock( '@wordpress/i18n', () => ( {
	__: jest.fn( ( text ) => text ),
} ) );

import { registerPlugin } from '@wordpress/plugins';

jest.mock( '@wordpress/plugins', () => ( {
	registerPlugin: jest.fn( ( name: string, settings ) => null ),
} ) );

describe( 'addEditorTools', () => {
	afterEach( () => {
		jest.resetAllMocks();
		jest.restoreAllMocks();
	} );

	it( 'should add editor tools', () => {
		const mockDocument = new DOMParser().parseFromString(
			`<html><body>
                <div class="editor-document-tools">
                    <div class="editor-document-tools__left"></div>
                </div>
            </body> </html>`,
			'text/html'
		);
		const onClick = jest.fn();

		addEditorTools( onClick, mockDocument );

		expect( onClick ).not.toHaveBeenCalled();
		expect(
			mockDocument.querySelectorAll(
				'.editor-document-tools__left .tec-editor-tool'
			).length
		).toBe( 0 );
		expect( registerPlugin ).toHaveBeenCalledWith( 'tec-editor-tools', {
			render: expect.any( Function ),
		} );

		const registerPluginMock = registerPlugin as jest.Mock;
		const registerPluginMockCallSettings = registerPluginMock.mock
			.calls[ 0 ][ 1 ] as { render: Function };
		const renderEditorTools = registerPluginMockCallSettings.render;

		const renderedEditorTools = renderEditorTools();

		expect( renderedEditorTools ).toBeNull();
		expect(
			mockDocument.querySelectorAll(
				'.editor-document-tools__left .tec-editor-tool'
			).length
		).toBe( 1 );
		expect(
			mockDocument.querySelector(
				'.editor-document-tools__left .tec-editor-tool'
			).outerHTML
		).toMatchSnapshot();

		const secondRenderedEditorTools = renderEditorTools();

		expect( secondRenderedEditorTools ).toBeNull();
		expect(
			mockDocument.querySelectorAll(
				'.editor-document-tools__left .tec-editor-tool'
			).length
		).toBe( 1 );

		const button = mockDocument.querySelector(
			'.editor-document-tools__left .tec-editor-tool'
		) as HTMLElement;
		button.click();
		expect( onClick ).toHaveBeenCalledTimes( 1 );
	} );

	it( 'should not add any button if the document does not have the editor tools', () => {
		const mockDocument = new DOMParser().parseFromString(
			`<html><body>
                <div class="editor-document-tools">
                </div>
            </body> </html>`,
			'text/html'
		);
		const onClick = jest.fn();

		addEditorTools( onClick, mockDocument );

		expect( onClick ).not.toHaveBeenCalled();
		expect(
			mockDocument.querySelectorAll(
				'.editor-document-tools__left .tec-editor-tool'
			).length
		).toBe( 0 );
		expect( registerPlugin ).toHaveBeenCalledWith( 'tec-editor-tools', {
			render: expect.any( Function ),
		} );

		const registerPluginMock = registerPlugin as jest.Mock;
		const registerPluginMockCallSettings = registerPluginMock.mock
			.calls[ 0 ][ 1 ] as { render: Function };
		const renderEditorTools = registerPluginMockCallSettings.render;

		const renderedEditorTools = renderEditorTools();

		expect( renderedEditorTools ).toBeNull();
		expect(
			mockDocument.querySelectorAll(
				'.editor-document-tools .tec-editor-tool'
			).length
		).toBe( 0 );
	} );
} );
