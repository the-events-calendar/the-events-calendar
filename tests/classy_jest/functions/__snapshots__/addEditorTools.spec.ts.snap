// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addEditorTools should add editor tools 1`] = `"<button class="tec-editor-tool tec-editor-tool--preview button" type="button" data-toolbar-item="true"><span class="dashicons dashicons-visibility"></span>Visual</button>"`;
